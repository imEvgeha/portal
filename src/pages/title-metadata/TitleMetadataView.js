import React, {useState} from 'react';
import PropTypes from 'prop-types';
import Button from '@atlaskit/button';
import {toggleRefreshGridData} from '@vubiquity-nexus/portal-ui/lib/grid/gridActions';
import {URL} from '@vubiquity-nexus/portal-utils/lib/Common';
import {connect} from 'react-redux';
import TitleCreate from '../legacy/containers/metadata/dashboard/components/TitleCreateModal'; // replace with new component
import CatalogueOwner from './components/catalogue-owner/CatalogueOwner';
import TitleMetadataHeader from './components/title-metadata-header/TitleMetadataHeader';
import TitleMetadataTable from './components/title-metadata-table/TitleMetadataTable';
import {CREATE_NEW_TITLE, SYNC_LOG, DEFAULT_CATALOGUE_OWNER} from './constants';
import './TitleMetadataView.scss';

export const TitleMetadataView = ({history, toggleRefreshGridData}) => {
    const [showModal, setShowModal] = useState(false);
    const [catalogueOwner, setCatalogueOwner] = useState({
        tenantCode: DEFAULT_CATALOGUE_OWNER,
    });

    const closeModalAndRefreshTable = () => {
        setShowModal(false);
        toggleRefreshGridData(true);
    };

    const changeCatalogueOwner = owner => {
        setCatalogueOwner(prevState => {
            return {
                ...prevState,
                tenantCode: owner,
            };
        });
    };

    return (
        <div className="nexus-c-title-metadata">
            <TitleMetadataHeader>
                <CatalogueOwner setCatalogueOwner={changeCatalogueOwner} />
                <Button
                    className="nexus-c-title-metadata__create-btn"
                    appearance="primary"
                    onClick={() => setShowModal(true)}
                >
                    {CREATE_NEW_TITLE}
                </Button>
                <Button
                    className="nexus-c-title-metadata__sync-btn"
                    appearance="subtle"
                    onClick={() => history.push(URL.keepEmbedded('/metadata/sync-log'))}
                >
                    {SYNC_LOG}
                </Button>
            </TitleMetadataHeader>
            <TitleMetadataTable history={history} catalogueOwner={catalogueOwner} />
            <TitleCreate
                display={showModal}
                toggle={closeModalAndRefreshTable}
                tenantCode={catalogueOwner.tenantCode}
            />
        </div>
    );
};

const mapDispatchToProps = dispatch => ({
    toggleRefreshGridData: payload => dispatch(toggleRefreshGridData(payload)),
});

TitleMetadataView.propTypes = {
    history: PropTypes.object,
    toggleRefreshGridData: PropTypes.func,
};

TitleMetadataView.defaultProps = {
    history: {},
    toggleRefreshGridData: () => null,
};

export default connect(null, mapDispatchToProps)(TitleMetadataView);
