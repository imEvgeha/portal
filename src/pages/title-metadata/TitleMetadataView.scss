.nexus-c-title-metadata {
    height: 100%;
    display: flex;
    flex-direction: column;
}
