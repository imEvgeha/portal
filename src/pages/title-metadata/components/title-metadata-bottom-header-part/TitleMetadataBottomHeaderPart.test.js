import React from 'react';
import {shallow} from 'enzyme';
import configureStore from 'redux-mock-store';
import {TitleMetadataView} from '../../TitleMetadataView';

describe('TitleMetadataBottomHeaderPart', () => {
    let wrapper = null;
    const mockStore = configureStore();
    const store = mockStore({
        titleMetadata: {
            filter: {
                filterModel: {},
                sortModel: null,
                columnState: [],
            },
        },
    });

    beforeAll(() => {
        wrapper = shallow(<TitleMetadataView store={store} />);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render bottom container for title metadata header', () => {
        expect(wrapper.find('.nexus-c-title-metadata-header__bottom')).toHaveLength(1);
    });
});
