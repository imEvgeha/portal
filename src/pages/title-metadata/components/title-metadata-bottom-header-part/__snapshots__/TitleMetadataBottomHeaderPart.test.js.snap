// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleMetadataBottomHeaderPart should match snapshot 1`] = `
<div
  className="nexus-c-title-metadata"
>
  <TitleMetadataHeader
    label="Title Metadata"
  >
    <div
      className="row"
    >
      <div
        className="col-4"
      >
        <div
          className="nexus-c-title-metadata-header__label"
        >
          Title Metadata
        </div>
      </div>
      <div
        className="col-4 d-flex justify-content-center"
      >
        <TabMenu
          activeIndex={0}
          className="nexus-c-title-metadata__tab-menu"
          id={null}
          model={
            Array [
              Object {
                "label": "Repository",
                "value": "repository",
              },
              Object {
                "label": "Sync Status",
                "value": "syncLog",
              },
              Object {
                "label": "Upload Log",
                "value": "uploadLog",
              },
            ]
          }
          onTabChange={[Function]}
          style={null}
        />
      </div>
      <div
        className="col-4"
      >
        <Connect(RepositorySelectsAndButtons)
          applyPredefinedTableView={[Function]}
          changeCatalogueOwner={[Function]}
          columnApi={null}
          getNameOfCurrentTab={[Function]}
          gridApi={null}
          lastStoredFilter={
            Object {
              "label": undefined,
            }
          }
          setBlockLastFilter={[Function]}
          setShowModal={[Function]}
          setUserDefinedGridState={[Function]}
          userDefinedGridStates={Array []}
          username=""
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <Connect(TitleMetadataBottomHeaderPart)
          className="nexus-c-title-metadata-header__bottom"
          isItTheSameTab={[Function]}
          showError={[Function]}
          showSuccess={[Function]}
          uploadHandler={[Function]}
        />
      </div>
    </div>
  </TitleMetadataHeader>
  <Connect(TitleMetadataTable)
    catalogueOwner={
      Object {
        "tenantCode": "vu",
      }
    }
    className="nexus-c-title-metadata__table"
    columnApi={null}
    gridApi={null}
    history={Object {}}
    setColumnApi={[Function]}
    setGridApi={[Function]}
  />
  <Connect(ComposedComponent)
    display={false}
    redirectToV2={true}
    tenantCode="vu"
    toggle={[Function]}
  />
</div>
`;
